addons/sale_purchase/models/sale_order.py

```diff
@@ -23,8 +23,8 @@ def _action_confirm(self):
             order.order_line.sudo()._purchase_service_generation()
         return result
 
-    def action_cancel(self):
-        result = super(SaleOrder, self).action_cancel()
+    def _action_cancel(self):
+        result = super(SaleOrder, self)._action_cancel()
```

Comment:
```suggestion
        result = super()._action_cancel()
```
let's clean this while we're on it :eyes: